contentcuration/contentcuration/assets/shared/views/PrimaryDialog.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <VDialog
+    v-model="open"
+    :disabled="disabled"
+    attach="body"
+    maxWidth="290"
+  >
+    <VCard>
+      <VCardTitle
+        v-if="title"
+      >
+        {{ title }}
+      </VCardTitle>
+      <VCardText>
+        <slot></slot>
+      </VCardText>
+      <VCardActions>
+        <slot name="actions"></slot>
+      </VCardActions>
+    </VCard>
+  </VDialog>
+</template>
+
+<script>
+
+  import Vue from 'vue';
+
+  const modalTracker = Vue.observable({
+    open: false,
+    name: '',
+  });
+
+  export default {
+    name: 'PrimaryDialog',
+    props: {
+      // Indicates whether the instantiator of this
+      // modal thinks that the modal should be open, but
+      // makes no guarantees.
+      value: {
+        type: Boolean,
+        required: true,
+      },
+      title: {
+        type: String,
+        default: '',
+      },
+    },
+    computed: {
+      disabled() {
+        return modalTracker.open && modalTracker.name !== this.$parent.$options.name;
```

Comment:
Disable the dialog in the case that another primary dialog is currently open!

Note that this will actually break in the case that the primary dialog is used twice in the same component, so I'll have to revisit this logic and update with an instance id instead.